client/apiversion_test.go

```diff
@@ -0,0 +1,164 @@
+package client_test
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	gc "gopkg.in/check.v1"
+)
+
+type versionHandler struct {
+	authBody string
+	port     string
+}
+
+type makeServiceURLTest struct {
+	serviceType string
+	version     string
+	parts       []string
+	success     bool
+	URL         string
+	err         string
+}
+
+func (s *localLiveSuite) makeServiceURLTests() []makeServiceURLTest {
+	return []makeServiceURLTest{
+		{
+			serviceType: "compute",
+			version:     "v2.1",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/v2.1/foo/bar/",
+		},
+		{
+			serviceType: "compute",
+			version:     "v2.0",
+			parts:       []string{},
+			success:     true,
+			URL:         "http://localhost:%s/v2.0",
+		},
+		{
+			serviceType: "compute",
+			version:     "v2.0",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/v2.0/foo/bar/",
+		},
+		{
+			serviceType: "compute",
+			version:     "v2",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/v2.1/foo/bar/",
+		},
+		{
+			serviceType: "object-store",
+			version:     "",
+			parts:       []string{"foo", "bar"},
+			success:     true,
+			URL:         "http://localhost:%s/swift/v1/foo/bar",
+		},
+		{
+			serviceType: "object-store",
+			version:     "v2.0",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/swift/v1/foo/bar/",
```

Comment:
The code comments say that container endpoints have no versioning capability and yet there's a "v1" in the expected URL. Isn't that versioning?  It seems counter intuitive that we can ask for a v2.0 endpoint and get back a v1 URL when that would fail for compute.